test/terra/backends/aer_simulator/test_shot_branching.py

```diff
@@ -595,6 +595,10 @@ def test_shot_branching_kraus_circuit_noise(self, method, device):
     # ---------------------------------------------------------------------
     # Test conditional
     # ---------------------------------------------------------------------
+    @unittest.skipUnless(
```

Comment:
Can these tests be fixed to work with the new `if` mechanism in qiskit 2.0?

Reply:
To some extent, yes. From what I have seen, the new `if` mechanism is only supported by some simulator methods (for example, `statevector` doesn't support it). I think that moving forward the tests would have to be refactored and only test conditionals with the supported simulator methods, but this is a bit tricky at the moment because we still want to support the "old" path, and there are already some tests for the new `if`, so I figured that the best intermediate solution would be the conditional skip. I can open an issue to keep track of this.

Reply:
An issue seems like the correct way to proceed, thank you.